chef/cookbooks/barclamp/spec/libraries/barclamp_library_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "spec_helper"
+
+require_relative "../../libraries/barclamp_library"
+
+describe BarclampLibrary::Barclamp::Inventory::Disk do
+
+  before(:each) do
+    @chef_run = ::ChefSpec::Runner.new
+    @node = @chef_run.node
+    @node.default[:block_device] = {
+      dm0: { removable: "0" },
+      xvd1: { removable: "0" },
+      xvd2: { removable: "1" }
+    }
+  end
+
+  it "unclaimed method returns the proper number of unclaimed devices" do
+    a = BarclampLibrary::Barclamp::Inventory::Disk
+    expect(a).to receive(:`).exactly(5).times.and_return(`exit 1`)
+    expect(::File).to receive(:exist?).with("/sys/block/dm0/dm/uuid").and_return(true)
+    expect(::File).to receive(:exist?).with("/sys/block/xvd2/dm/uuid").and_return(false)
+    expect(::File).to receive(:exist?).with("/sys/block/sr0/dm/uuid").and_return(false)
+    expect(::File).to receive(:open).exactly(1).times.with(
+      "/sys/block/dm0/dm/uuid"
+    ).and_yield(StringIO.new("mpath-test"))
+    # return holders
+    expect(::Dir).to receive(:entries).exactly(5).times.and_return([])
+    expect(a.unclaimed(@node).length).to eq(3)
+  end
```

Comment:
Not the one in the comment exactly no, I did not use the comment as guide for doing the tests.